FiveCalls/FiveCalls/CallScriptViewController.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  CallScriptViewController.swift
+//  FiveCalls
+//
+//  Created by Patrick McCarron on 2/3/17.
+//
+
+import UIKit
+import CoreLocation
+
+class CallScriptViewController : UIViewController {
+    
+    var issuesManager: IssuesManager!
+    var issue: Issue!
+    var contact: Contact!
+    var contacted: Set<Contact> = []
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    @IBOutlet weak var resultUnavailableBtn: UIButton!
+    @IBOutlet weak var resultVoicemailBtn: UIButton!
+    @IBOutlet weak var resultContactedBtn: UIButton!
+    @IBOutlet weak var resultSkipBtn: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .action, target: nil, action: nil)
+        
+        tableView.estimatedRowHeight = 100
+        tableView.rowHeight = UITableViewAutomaticDimension
+    }
+    
+    func callButtonPressed(_ button: UIButton) {
+        if let dialURL = URL(string: "telprompt:\(contact.phone)") {
+            UIApplication.shared.open(dialURL) { success in
+                //Log the result
+            }
+        }
+    }
+
+    @IBAction func resultButtonPressed(_ button: UIButton) {
+        switch button {
+        case resultContactedBtn:
+            print("call connected, log this")
+            break
+        case resultSkipBtn:
+            print("call skipped, log this")
+            break
+        case resultVoicemailBtn:
+            print("got voicemail, log this")
+            break
+        case resultUnavailableBtn:
+            print("unavailable, log this")
+            break
+        default:
+            print("unknown button pressed")
+        }
+        
+        // Struct passing around problems. This needs to be refactored / cleaned up.
+        contact.hasContacted = true
+        let oldContact: Contact! = contact
+        for i in 0..<issuesManager.issuesList!.issues.count {
+            if issuesManager.issuesList!.issues[i].id == self.issue.id {
+                for j in 0..<issuesManager.issuesList!.issues[i].contacts.count {
+                    if issuesManager.issuesList!.issues[i].contacts[j].id == oldContact.id {
+                        issuesManager.issuesList!.issues[i].contacts[j].hasContacted = true
+                    }
+                    if issuesManager.issuesList!.issues[i].contacts[j].hasContacted == false {
+                        contact = issuesManager.issuesList!.issues[i].contacts[j]
+                        tableView.reloadData()
+                    }
+                }
+            }
+        }
+        if (contact.hasContacted) {
+            for i in 0..<issuesManager.issuesList!.issues.count {
+                if issuesManager.issuesList!.issues[i].id == self.issue.id {
+                    issuesManager.issuesList!.issues[i].madeCalls = true
+                }
+            }
```

Comment:
Another thing to consider is we will need some form of persistence soon, right now we use `UserDefaults` for your location settings which seems appropriate, but when it comes to saving these stats locally, we'll have to probably put them somewhere else (I was thinking of using @nickoneill's [Pantry](https://github.com/nickoneill/Pantry)).  However if we decided to just leverage a data store we could go with Realm or CoreData also. Food for thought.